### PR TITLE
Upgrade nvcomp to 4.0.1

### PR DESCRIPTION
**Pull request by vuule**
## Description
Upgrade to nvcomp 4.0. As part of that change we need to refactor the `nvcomp.cmake` to be aware of the new CMake files and targets that nvcomp generates.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rapids-cmake/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by robertmaynard** [comment]
URL form for nvcomp 4.0 has changed we need `nvcomp-linux-sbsa-4.0.0-cuda12.5.tar.gz` instead of `nvcomp_4.0.0_SBSA_12.x.tgz`

Likewise for x86 which is now `nvcomp-linux-x86_64-4.0.0-cuda12.5.tar.gz`

**Comment by robertmaynard** [comment]
Going to look at the changes required in rapids-cmake to support these new url strings later today

**Comment by robertmaynard** [comment]
More changes are required as the nvcomp 4.0 cmake package has a different set of files which needs to be patched for a proper install layout

**Comment by robertmaynard** [comment]
Waiting on https://github.com/rapidsai/cudf/pull/16076/files being ready to merge

**Comment by robertmaynard** [comment]
Blocked due to nvcomp now having a hard libcuda.so dependency.

File: rapids-cmake/cpm/nvcomp.cmake [hunk 1]
```diff
@@ -187,15 +202,13 @@ function(rapids_cpm_nvcomp)
...
-    add_library(nvcomp::nvcomp ALIAS nvcomp)
-  endif()
-  if(NOT TARGET nvcomp::nvcomp_gdeflate AND TARGET nvcomp_gdeflate)
-    add_library(nvcomp::nvcomp_gdeflate ALIAS nvcomp_gdeflate)
-  endif()
-  if(NOT TARGET nvcomp::nvcomp_bitcomp AND TARGET nvcomp_bitcomp)
-    add_library(nvcomp::nvcomp_bitcomp ALIAS nvcomp_bitcomp)
-  endif()
+  set(nvcomp_possible_target_names nvcomp nvcomp_bitcomp nvcomp_cpu nvcomp_cpu_static
+                                   nvcomp_device_static nvcomp_gdeflate nvcomp_static)
```
Comment (bdice) [inline rapids-cmake/cpm/nvcomp.cmake:206]: `nvcomp_bitcomp` and `nvcomp_gdeflate` are no longer separate libraries in nvcomp 4.0.1. Should we keep them in this list for historical / backwards-compatibility reasons or should we remove them?

**Review by bdice — APPROVED** [review]
One question, otherwise this looks good to me.

File: rapids-cmake/cpm/nvcomp.cmake (hunk 1 above)
Comment (robertmaynard) [reply rapids-cmake/cpm/nvcomp.cmake]: Removed usages

**Pull request merged**